### PR TITLE
Improve developer documentation

### PR DESCRIPTION
**Pull request by subhramit**
## Documentation improvements
Refs https://github.com/JabRef/jabref/pull/11507 which had to be closed due to https://github.com/JabRef/jabref/pull/11507#issuecomment-2268025339. This PR adds the remaining non-conflicting changes:
- Add link to developer FAQs in first time contribution greeting.
- Grammar fix in `faq.md`.
- Add some pointers about reaching IntlliJ Settings in M2 Macs while setting up the development environment.

### Mandatory checks

- [ ] ~Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)~
- [ ] ~Tests created for changes (if applicable)~
- [ ] ~Manually tested changed features in running JabRef (always required)~
- [ ] ~Screenshots added in PR description (for UI changes)~
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: docs/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.md [hunk 1]
```diff
@@ -9,7 +9,8 @@ nav_order: 13
 Contributions to JabRef's source code need to have a code formatting that is consistent with existing source code. For that purpose, JabRef provides code-style and check-style definitions.
 
 Install the [CheckStyle-IDEA plugin](http://plugins.jetbrains.com/plugin/1065?pr=idea), it can be found via the plug-in repository:
-Navigate to **File > Settings... > Plugins"**.
+Navigate to **File > Settings... > Plugins"**.<br>
```
Comment (koppor) [inline docs/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.md:12]: I hink, the line "On the top" will directly come after `"File".` This is not intended.

File: docs/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-12-build.md [hunk 2]
```diff
@@ -8,7 +8,8 @@ nav_order: 12
 
 ## Ensure that JDK 21 is available to IntelliJ
 
-Ensure you have a Java 21 SDK configured by navigating to **File > Project Structure... > Platform Settings > SDKs**.
+Ensure you have a Java 21 SDK configured by navigating to **File > Project Structure... > Platform Settings > SDKs**.<br>
+**Note:** In some MacBooks, `Settings` can be found at the "IntelliJ" button of the app menu instead of at "File".
```
Comment (subhramit) [inline docs/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-12-build.md:12]: I will have to change `Settings` to `Project Structure`

File: docs/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.md (hunk 1 above)
Comment (subhramit) [reply docs/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.md:12]: The flow in Windows/old macs: `File`->`Settings`->`Plugins`->`Marketplace`...
The flow in M2 macs: `IntelliJ`->`Settings`->`Plugins`->`Marketplce`...
Only change is the first action's location, which had to be alternatively pointed out for m2 macs below, rest is the same. So I think the line in the middle should stay.

Comment (subhramit) [reply docs/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.md:12]: OK I think I get what you mean now, just checked the preview.
Yes, it's better if we move the middle line (13) to after 16.

File: docs/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.md [hunk 3]
```diff
@@ -12,7 +12,8 @@ Install the [CheckStyle-IDEA plugin](http://plugins.jetbrains.com/plugin/1065?pr
 Navigate to **File > Settings... > Plugins"**.
 On the top, click on "Marketplace".
 Then, search for "Checkstyle".
-Click on "Install" choose "CheckStyle-IDEA".
+Click on "Install" choose "CheckStyle-IDEA".<br>
```
Comment (koppor) [inline docs/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.md:15]: `<br>` is like <kbd>Shift</kbd>+<kbd>Enter </kbd>in word.

Did you experiment with 

- a new paragraph
- Using `>` (and a new paragraph)
- Using the "note" mark? (This is too heavyweight - see https://just-the-docs.com/#getting-started). However, it is exactly meant for notes.

Comment (subhramit) [reply docs/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.md:15]: Today I learnt: Two spaces in markdown adds a new line.
https://www.markdownguide.org/basic-syntax/#line-break-best-practices

Comment (koppor) [reply docs/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.md:15]: > Today I learnt: Two spaces in markdown adds a new line. [markdownguide.org/basic-syntax/#line-break-best-practices](https://www.markdownguide.org/basic-syntax/#line-break-best-practices)

But very hidden in the editor

Also with automatic removel of white spaces at the end. Therefore, we have the special editor config: https://github.com/JabRef/jabref/blob/95a81e8bd7837a1b0b7fa4ae252ea3b816ad8cb9/.editorconfig#L12

Comment (koppor) [reply docs/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.md:15]: Ah, still: two spaces are shift+enter in Word. Which is a no-go.

**Review by koppor — APPROVED** [review]
OK, I don't care about clean Markdown this time. Will be fixed in future work.

**Pull request merged**